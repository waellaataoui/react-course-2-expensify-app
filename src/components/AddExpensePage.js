import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";
export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push("/dashboard");
  };
  render() {
    return (
      <div>
        <h1>
          Add ExpenseForm
          <ExpenseForm onSubmit={this.onSubmit} />
        </h1>
      </div>
    );
  }
}
const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});
export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);
