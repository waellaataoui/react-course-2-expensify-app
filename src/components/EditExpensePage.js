import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense } from "../actions/expenses";
import { removeExpense } from "../actions/expenses";
export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    // this.props.editExpense(this.props.match.params.id, expense);
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  onClick = () => {
    this.props.removeExpense(this.props.expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <ExpenseForm onSubmit={this.onSubmit} expense={this.props.expense} />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(
    (expense) => expense.id === props.match.params.id
  )
});
const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, updates) => dispatch(editExpense(id, updates)),
  removeExpense: (expense) => dispatch(removeExpense(expense))
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
