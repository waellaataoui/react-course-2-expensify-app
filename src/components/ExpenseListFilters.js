import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate
} from "../actions/filters";
export class ExpenseListFilters extends React.Component {
  state = {
    calandarFocus: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calandarFocus) => {
    this.setState(() => ({
      calandarFocus
    }));
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    e.target.value === "date"
      ? this.props.sortByDate()
      : this.props.sortByAmount();
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="sid"
          endDate={this.props.filters.endDate}
          endDateId="eid"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calandarFocus}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  filters: state.filters
});
const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
});
export default connect(mapStateToProps)(ExpenseListFilters);
