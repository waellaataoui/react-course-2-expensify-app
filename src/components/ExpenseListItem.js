import moment from "moment";
import numeral from "numeral";
import React from "react";
import { Link } from "react-router-dom";
const ExpenseListItem = (props) => {
  return (
    <Link className="list__item" to={`/edit/${props.expense.id}`}>
      <div>
        <h3 className="list__item__title">{props.expense.description}</h3>
        <span className="list__item__subtitle">
          {moment(props.expense.createdAt).format("MMMM Do, YYYY")}
        </span>
      </div>

      <h3>{numeral(props.expense.amount / 100).format("$0,0.00")}</h3>
    </Link>
  );
};
export default ExpenseListItem;
