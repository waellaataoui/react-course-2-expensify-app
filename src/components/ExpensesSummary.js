import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";
export const ExpensesSummary = (props) => (
  <div>
    <h1>
      viewing {props.expenseCount}{" "}
      {props.expenseCount === 1 ? "expense" : "expenses"} totalling{" "}
      {numeral(props.expensesTotal / 100).format("$0,0.00")}
    </h1>
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses)
  };
};
export default connect(mapStateToProps)(ExpensesSummary);
