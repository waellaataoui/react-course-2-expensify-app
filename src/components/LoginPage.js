import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
export const LoginPage = ({ startLogin }) => (
  <div>
    <h3>Budget App</h3>
    <p>It's time to get your expenses under control.</p>
    <button onClick={startLogin}>Login with Google</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});
export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
