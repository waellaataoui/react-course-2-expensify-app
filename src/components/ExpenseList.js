import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import ExpenseListItem from "./ExpenseListItem";
export const ExpenseList = (props) => (
  <div className="list content-container">
    <div className="list__header">
      <p>Expense</p>
      <p> Amount</p>
    </div>
    {props.expenses.length === 0 ? (
      <p className="list__msg">No expenses</p>
    ) : (
      props.expenses.map((expense) => (
        <ExpenseListItem key={expense.id} expense={expense} />
      ))
    )}
  </div>
);
const mapSateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)
});
export default connect(mapSateToProps)(ExpenseList);
