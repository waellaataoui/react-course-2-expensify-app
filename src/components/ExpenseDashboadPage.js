import React from "react";

import ExpensesSummary from "../components/ExpensesSummary";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
