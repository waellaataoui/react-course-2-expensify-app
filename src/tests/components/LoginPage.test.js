import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";
test("should render login page without errors", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});
test("should call start login on button click ", () => {
  const startLoginSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLoginSpy}></LoginPage>);
  wrapper.find("button").simulate("click");
  expect(startLoginSpy).toHaveBeenCalled();
});
