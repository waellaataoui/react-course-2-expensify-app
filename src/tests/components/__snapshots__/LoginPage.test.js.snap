// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page without errors 1`] = `
<div>
  <h3>
    Budget App
  </h3>
  <p>
    It's time to get your expenses under control.
  </p>
  <button>
    Login with Google
  </button>
</div>
`;
